### ISSUE TITLE
[api/spec] Incomplete ConfigAccessor description

### ISSUE THREAD

**Issue by jmesnil** [eclipse/ConfigJSR#70]
I'm integrating the ConfigAccessor to my config JSR implementation and the following concepts from ConfigAccessor spec are not explained or specified in https://github.com/eclipse/ConfigJSR/blob/master/spec/src/main/asciidoc/configaccessor.asciidoc#configaccessor:

* `withCurrentProjectStage(boolean with)` does not define what a project state is.
* `strictly(boolean strictly)` has a local link for `property value resolution sequence` that does not exist.
* `parameterizedBy(String propertyName)` has a local link for `ConfigResolver.getPropertyAwarePropertyValue` that does not exist

There is also disprecancy between the spec and the API.
The `ConfigAccessor` does not define some methods specified in the spec (e.g. `withCurrentProjectStage`, `strictly`) while some in the interface are not mentioned (e.g. `withLookupChain`).

The Javadoc of `withLookupChain` mentions a `javaconfig.projectStage` property that is not defined anywhere. Is the name of this property significant (I assume it is since it starts with `javaconfig`).

**Comment by struberg** [comment]
Right, withProjectStage should be removed. It's a left-over from DeltaSpike.
We could probably introduce a common property `javax.config.projectStage` or so.

But in that case you would use `withLookupChain("${javax.config.projectStage}")` to make Config use a ".Production"  or ".Development" postfix lookup.

**Comment by struberg** [comment]
@jmesnil the spec paper should now be in a much better shape. Can you please review if all is resolved? txs!
